src/freenet/clients/http/FProxyToadlet.java

```diff
@@ -409,46 +409,88 @@ private static boolean horribleEvilHack(Bucket data) throws IOException {
 			byte[] buf = new byte[sz];
 			// FIXME Fortunately firefox doesn't detect RSS in UTF16 etc ... yet
 			is.readFully(buf);
-			/**
-		 * Look for any of the following strings:
-		 * <rss
-		 * &lt;feed
-		 * &lt;rdf:RDF
-		 *
-		 * If they start at the beginning of the file, or are preceded by one or more &lt;! or &lt;? tags,
-		 * then firefox will read it as RSS. In which case we must force it to be downloaded to disk.
-		 */
-			if(checkForString(buf, "<rss"))
-				return true;
-			if(checkForString(buf, "<feed"))
-				return true;
-			if(checkForString(buf, "<rdf:RDF"))
-				return true;
+			return isSniffedAsFeed(buf);
 		}
 		finally {
 			Closer.close(is);
 		}
-		return false;
 	}
 
-	/** Scan for a US-ASCII (byte = char) string within a given buffer of possibly binary data */
-	private static boolean checkForString(byte[] buf, String find) {
-		int offset = 0;
-		int bufProgress = 0;
-		while(offset < buf.length) {
-			byte b = buf[offset];
-			if(b == find.charAt(bufProgress)) {
-				bufProgress++;
-				if(bufProgress == find.length()) return true;
-				offset++;
-			} else {
-				if(bufProgress == 0)
-					offset++; // Try the next byte.
-				else
-					bufProgress = 0; // Reset to the first char of the keyword.
+	/**
```

Comment:
Oh, and can you move this to its own file? `FproxyToadlet` doesn’t really need to grow… :)